create_builder.go

```diff
@@ -40,16 +42,94 @@ func (c *Client) CreateBuilder(ctx context.Context, opts CreateBuilderOptions) e
 		return errors.Wrap(err, "fetch build image")
 	}
 
+	builder, err := c.createBaseBuilder(ctx, opts, baseImage)
+	if err != nil {
+		return errors.Wrap(err, "failed to create builder")
+	}
+
+	err = c.addBuildpacksToBuilder(ctx, opts, builder)
+	if err != nil {
+		return errors.Wrap(err, "failed to add buildpacks to builder")
+	}
+
+	builder.SetOrder(opts.Config.Order)
+	builder.SetStack(opts.Config.Stack)
+
+	return builder.Save(c.logger)
+}
+
+func validateBuilderConfig(conf pubbldr.Config) error {
```

Comment:
Move below `*Client` methods.

Reply:
[not blocking] Also, this could be public and tested even though its internal? It could also be a method of `Config` ie `Config.Validate() error`?

Reply:
I moved it to be a method of the `StackConfig`. In terms of design, though, would it be better to have it there, or not, given that that's adding funcs to data objects?

Reply:
@dfreilich I would be in favor of not adding logic to data types but everything else in this PR is a great improvement. This is not something that should block this PR and we can change it later.